sdk/dotnet/Pulumi/Deployment/GrpcEngine.cs

```diff
@@ -12,7 +13,9 @@ internal class GrpcEngine : IEngine
 
         public GrpcEngine(string engine)
         {
-            this._engine = new Engine.EngineClient(new Channel(engine, ChannelCredentials.Insecure));
+            var maxRpcMessageSize = 400 * 1024 * 1024;
```

Comment:
Perhaps add a comment explaining this constant?  Here and in all the other similar locations.